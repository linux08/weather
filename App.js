import React from 'react';
import { Platform, StatusBar, StyleSheet, View, Text } from 'react-native';
import { Provider } from 'react-redux';
import store from './store';
import { AppLoading, Asset, Font, Icon } from 'expo';
import Navigator from './navigation/Navigator';


const Store = store();

export default class App extends React.Component {
  state = {
    isLoadingComplete: false,
  };

  async componentDidMount() {
    await this.loadResourcesAsync();
    this.setState({
      isLoadingComplete: true,
    });
  }

  loadResourcesAsync = async () => {
    Font.loadAsync({
      // This is the font that we are using for our tab bar
      ...Icon.Ionicons.font,
    });
  };

  handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };


  render() {
    const { isLoadingComplete } = this.state;
    if (!isLoadingComplete) {
      return (
        <AppLoading
          startAsync={this.loadResourcesAsync}
          onError={this.handleLoadingError}
          onFinish={this.handleFinishLoading}
        />
      );
    }
    return (
      <Provider store={Store}>
        <View style={styles.container}>
          <Navigator />
        </View>
      </Provider>

    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
