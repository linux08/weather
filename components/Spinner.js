import React from 'react';
import { View, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';

const styles = {
    spinnerStyle: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
};

const Spinner = ({ size, color }) => (
    <View style={styles.spinnerStyle}>
        <ActivityIndicator size={size || 'large'} color={color || '#2F3CA3'} />
    </View>
);

Spinner.defaultProps = {
    size: 'large',
    color: '#2F3CA3',
};

Spinner.propTypes = {
    size: PropTypes.string,
    color: PropTypes.string,
};

export default Spinner;