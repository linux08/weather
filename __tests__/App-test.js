import React from 'react';
import App from '../App';
import Spinner from '../components/Spinner';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => { });
jest.mock('react-navigation-stack', () => { });
jest.mock('react-navigation', () => {
  return {
    createAppContainer: jest.fn().mockReturnValue(function NavigationContainer(props) { return null; }),
    createDrawerNavigator: jest.fn(),
    createBottomTabNavigator: jest.fn(),
    createMaterialTopTabNavigator: jest.fn(),
    createStackNavigator: jest.fn(),
    StackActions: {
      push: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/PUSH" })),
      replace: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/REPLACE" })),
    },
    NavigationActions: {
      navigate: jest.fn().mockImplementation(x => x),
    }
  }
});

jest.mock("NativeModules", () => ({
  UIManager: {
    RCTView: () => ({
      directEventTypes: {}
    })
  },
  KeyboardObserver: {},
  RNGestureHandlerModule: {
    attachGestureHandler: jest.fn(),
    createGestureHandler: jest.fn(),
    dropGestureHandler: jest.fn(),
    updateGestureHandler: jest.fn(),
    State: {},
    Directions: {}
  },
  PlatformConstants: {
    forceTouchAvailable: false,
  },
}));


describe('App snapshot', () => {
  jest.useFakeTimers();
  jest.setTimeout(30000);
  it('renders the Main screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
