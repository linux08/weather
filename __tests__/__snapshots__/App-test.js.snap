// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the Main screen 1`] = `null`;

exports[`App snapshot renders the SPinner 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2F3CA3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`App snapshot renders the Spinner 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2F3CA3"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `null`;
